core/src/test/scala/integration/kafka/api/ClassLoaderTest.scala

```diff
@@ -0,0 +1,291 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.api
+
+import java.io.Closeable
+import java.io.File
+import java.net.URL
+import java.net.URLClassLoader
+import java.util.Collections
+import java.util.Map
+import java.util.Properties
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.collection.mutable.Buffer
+
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.consumer.RoundRobinAssignor
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.clients.producer.internals.DefaultPartitioner
+import org.apache.kafka.common.config.SslConfigs
+import org.apache.kafka.common.config.types.Password
+import org.apache.kafka.common.protocol.SecurityProtocol
+import org.apache.kafka.common.security.auth.DefaultPrincipalBuilder
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.StringSerializer
+import org.apache.kafka.test.MockConsumerInterceptor
+import org.apache.kafka.test.MockProducerInterceptor
+import org.junit.{Before, After, Test}
+import org.junit.Assert._
+
+import kafka.server.KafkaConfig
+import kafka.utils.TestUtils
+
+class ClassLoaderTest extends IntegrationTestHarness with SaslSetup {
```

Comment:
@ijuma Thank you for the review. I thought it would be useful to have tests that put it all together and ran full end-to-end tests. But you are right, `AbstractConfigTest` would find any bugs, so this doesn't add enough value to justify the additional time spent in tests. I have removed the integration test.